botorch/models/transforms/outcome.py

```diff
@@ -631,3 +631,128 @@ def untransform_posterior(self, posterior: Posterior) -> Posterior:
+    ) -> Tuple[Tensor, Optional[Tensor]]:
+        r"""Bilog-transform outcomes.
+
+        Args:
+            Y: A `batch_shape x n x m`-dim tensor of training targets.
+            Yvar: A `batch_shape x n x m`-dim tensor of observation noises
+                associated with the training targets (if applicable).
+
+        Returns:
+            A two-tuple with the transformed outcomes:
+
+            - The transformed outcome observations.
+            - The transformed observation noise (if applicable).
+        """
+        Y_tf = torch.sign(Y) * torch.log(torch.abs(Y) + torch.tensor(1.0).to(Y))
+        outputs = normalize_indices(self._outputs, d=Y.size(-1))
+        if outputs is not None:
+            Y_tf = torch.stack(
+                [
+                    Y_tf[..., i] if i in outputs else Y[..., i]
+                    for i in range(Y.size(-1))
+                ],
+                dim=-1,
+            )
+        if Yvar is not None:
+            raise NotImplementedError(
+                "Tanh does not yet support transforming observation noise"
+            )
+        return Y_tf, Yvar
+
+    def untransform(
+        self, Y: Tensor, Yvar: Optional[Tensor] = None
+    ) -> Tuple[Tensor, Optional[Tensor]]:
+        r"""Un-transform bilog-transformed outcomes
+
+        Args:
+            Y: A `batch_shape x n x m`-dim tensor of power-transfomred targets.
+            Yvar: A `batch_shape x n x m`-dim tensor of power-transformed
+                observation noises associated with the training targets
+                (if applicable).
+
+        Returns:
+            A two-tuple with the un-transformed outcomes:
+
+            - The un-power transformed outcome observations.
+            - The un-power transformed observation noise (if applicable).
+        """
+        Y_utf = torch.sign(Y) * (torch.exp(torch.abs(Y)) - torch.tensor(1.0).to(Y))
+        outputs = normalize_indices(self._outputs, d=Y.size(-1))
+        if outputs is not None:
+            Y_utf = torch.stack(
+                [
+                    Y_utf[..., i] if i in outputs else Y[..., i]
+                    for i in range(Y.size(-1))
+                ],
+                dim=-1,
+            )
+        if Yvar is not None:
+            # TODO: Delta method, possibly issue warning
+            raise NotImplementedError(
+                "Bilog does not yet support transforming observation noise"
+            )
+        return Y_utf, Yvar
+
+    def untransform_posterior(self, posterior: Posterior) -> Posterior:
+        r"""Un-transform the tanh-transformed posterior.
+
+        Args:
+            posterior: A posterior in the tanh-transformed space.
+
+        Returns:
+            The un-transformed posterior.
+        """
+        if self._outputs is not None:
+            raise NotImplementedError(
+                "Bilog does not yet support output selection for untransform_posterior"
+            )
+        return TransformedPosterior(
```

Comment:
@Balandat, @saitcakmak  do we need a `mean_transform` and `variance_transform` here as well?